src/main/java/io/vertx/core/http/impl/Http1xClientConnection.java

```diff
@@ -898,6 +908,9 @@ synchronized void toWebSocket(
         !options.isSendUnmaskedFrames());
 
       WebSocketHandshakeInboundHandler handshakeInboundHandler = new WebSocketHandshakeInboundHandler(handshaker, ar -> {
+        if (timer > 0L) {
```

Comment:
@vietj should this possibly be changed to `if (timer >= 0L) {`?

When I set .setTimeout(timeout) in my websocket client, the very first  call to `vertx.setTimer(handshakeTimeout, id -> closeInternal())` returns `0`. 
Because of that, the timer is not cancelled after and closes connection - after a successful handshake.

I am able to avoid this by initializing the timer counter before websocket connection setup: `vertx.cancelTimer(vertx.setTimer(1, ignored -> {}));`.

Is there some other configuration that I might be missing, that helps to avoid this behaviour?